### PR TITLE
First time build fix

### LINKED ISSUE #38: Build instructions fail in key points
# Reference URL
- https://github.com/code4romania/impreuna-pentru-sanatate/wiki/Local-Development-Environment

# System:
- Windows 10 Professional x64
- Powershell

# Failure steps:
1. Docker relative path resolution
```
\war-support-un-acoperis\.docker\webserver\.docker\webserver: The system cannot find the path specified
```

2. NGINX configuration COPY 
```
 => ERROR [war-support-un-acoperis_webserver 2/6] COPY .docker/webserver/config/nginx/default /etc/nginx/sites-available/default                                                                                                                                          
 => ERROR [war-support-un-acoperis_webserver 3/6] COPY .docker/webserver/config/supervisor/supervisor.conf /etc/supervisor/conf.d/supervisor.conf                                                                                                                        
 => ERROR [war-support-un-acoperis_webserver 4/6] COPY .docker/webserver/config/nginx/nginx.conf /etc/nginx/nginx.conf
```

3. Artisan not installed
4. Migrations fail when updating with non-nullable column mutations

### PR DESCRIPTION
**Pull request by Cosmin-Parvulescu**
### What does it fix?

> Updated some settings that prevented the smooth build process of the solution.

Closes #38 

### How has it been tested?

Just manual e2e build process on Windows 10 Professional using Powershell

* **Wasn't tested on Linux**

### CODE REVIEW

**Review by beniamin — COMMENTED** [review]
The change of package-lock.json is intended? Or is just the result of npm install? If is the npm install, then you should use `npm ci` to install the packages with versions from package-lock.json and avoid updating it on every environment setup.

**Comment by Cosmin-Parvulescu** [comment]
@beniamin, [instructions](https://github.com/code4romania/impreuna-pentru-sanatate/wiki/Local-Development-Environment) should also be updated from:
```
cd app
php artisan migrate --seed
cp .env.example .env
php artisan key:generate
```
to
```
composer install
php artisan migrate --seed
cp .env.example .env
php artisan key:generate
```

But also check before, just have n = 1 for now, but I did manage to put the solution up from scratch just now so fresh eyes.

**Comment by beniamin** [comment]
> @beniamin, [instructions](https://github.com/code4romania/impreuna-pentru-sanatate/wiki/Local-Development-Environment) should also be updated from:
> 
> ```
> cd app
> php artisan migrate --seed
> cp .env.example .env
> php artisan key:generate
> ```
> 
> to
> 
> ```
> composer install
> php artisan migrate --seed
> cp .env.example .env
> php artisan key:generate
> ```
> 
> But also check before, just have n = 1 for now, but I did manage to put the solution up from scratch just now so fresh eyes.

Agree. I think env setup should be done before running artisan migrations.

File: app/Http/Middleware/SecurityHeaders.php [hunk 1]
```diff
@@ -16,11 +16,11 @@ class SecurityHeaders
      */
     public function handle($request, Closure $next)
     {
-        $allowedHosts = ['helpforhealth.local', 'impreunapentrusanatate.ro', 'dev.impreunapentrusanatate.ro'];
+        // $allowedHosts = ['helpforhealth.local', 'impreunapentrusanatate.ro', 'dev.impreunapentrusanatate.ro'];
```
Comment (Cosmin-Parvulescu) [inline app/Http/Middleware/SecurityHeaders.php:19]: Decided to just comment them out as these'll need to be updated with the new values

**Comment by Cosmin-Parvulescu** [comment]
> The change of package-lock.json is intended? Or is just the result of npm install? If is the npm install, then you should use `npm ci` to install the packages with versions from package-lock.json and avoid updating it on every environment setup.

👍

**Comment by beniamin** [comment]
I think env.example should be changed. It references some `helpforhealth` hosts. If this is not a problem for now, we can go on. Otherwise, the PR looks good to me.

**Comment by Cosmin-Parvulescu** [comment]
I'd rather leave that to you guys in another specific PR because on my side, my hosts is heavily protected and can't be modified so I ended up using just the ports and ngrok to test out the http and https connections. Furthermore, it's going to need changes across the app that should be aligned by name ```war-support-un-acoperis``` or just ```un-acoperis``` or however you decide. :)

Good luck!

**Comment by Cosmin-Parvulescu** [comment]
Re-opening as #39 is still red

**Comment by gheorghelupu17** [comment]
Handled much more elegantly in https://github.com/code4romania/war-support-un-acoperis/pull/39

**Pull request closed without merging**